br/pkg/lightning/backend/local/key_adapter.go

```diff
@@ -46,36 +47,41 @@ func reallocBytes(b []byte, n int) []byte {
 
 type noopKeyAdapter struct{}
 
-func (noopKeyAdapter) Encode(dst []byte, key []byte, _ int64) []byte {
+func (noopKeyAdapter) Encode(dst []byte, key []byte, _ []byte) []byte {
 	return append(dst, key...)
 }
 
 func (noopKeyAdapter) Decode(dst []byte, data []byte) ([]byte, error) {
 	return append(dst, data...), nil
 }
 
-func (noopKeyAdapter) EncodedLen(key []byte) int {
+func (noopKeyAdapter) EncodedLen(key []byte, _ []byte) int {
 	return len(key)
 }
 
 var _ KeyAdapter = noopKeyAdapter{}
 
 type dupDetectKeyAdapter struct{}
 
-func (dupDetectKeyAdapter) Encode(dst []byte, key []byte, rowID int64) []byte {
+func (dupDetectKeyAdapter) Encode(dst []byte, key []byte, rowID []byte) []byte {
 	dst = codec.EncodeBytes(dst, key)
-	dst = reallocBytes(dst, 8)
-	n := len(dst)
-	dst = dst[:n+8]
-	binary.BigEndian.PutUint64(dst[n:n+8], codec.EncodeIntToCmpUint(rowID))
+	dst = reallocBytes(dst, len(rowID)+2)
```

Comment:
what's the length of longest handle?

Reply:
It's max value of uint16(0xFFFF):

https://github.com/pingcap/tidb/blob/bc95a4f02270e10ccd0dba39b1c0513528bb2eba/tablecodec/tablecodec.go#L1635-L1641